conjure_oxide/src/rules/minion.rs

```diff
@@ -21,33 +23,57 @@ fn is_nested_sum(exprs: &Vec<Expr>) -> bool {
 /**
  * Helper function to get the vector of expressions from a sum (or error if it's a nested sum - we need to flatten it first)
  */
-fn sum_to_vector(expr: &Expr) -> Result<Vec<Expr>, RuleApplicationError> {
+fn sum_to_vector(expr: &Expr) -> Result<Vec<Expr>, ApplicationError> {
     match expr {
         Expr::Sum(_, exprs) => {
             if is_nested_sum(exprs) {
-                Err(RuleApplicationError::RuleNotApplicable)
+                Err(ApplicationError::RuleNotApplicable)
             } else {
                 Ok(exprs.clone())
             }
         }
-        _ => Err(RuleApplicationError::RuleNotApplicable),
+        _ => Err(ApplicationError::RuleNotApplicable),
     }
 }
 
+// /**
+//  * Convert an Eq to a conjunction of Geq and Leq:
+//  * ```text
+//  * a = b => a >= b && a <= b
+//  * ```
+//  */
+// #[register_rule(("Minion", 100))]
+// fn eq_to_minion(expr: &Expr, _: &Model) -> ApplicationResult {
+//     match expr {
+//         Expr::Eq(metadata, a, b) => Ok(Reduction::pure(Expr::And(
+//             metadata.clone(),
+//             vec![
+//                 Expr::Geq(metadata.clone(), a.clone(), b.clone()),
+//                 Expr::Leq(metadata.clone(), a.clone(), b.clone()),
+//             ],
+//         ))),
+//         _ => Err(ApplicationError::RuleNotApplicable),
+//     }
+// }
```

Comment:
Has commenting this out stopped `Eq` working? Just wanted to check what's going on.

Reply:
The Eq rule will interfere with other rules like the one that converts Min and the one that converts to SumGeq etc. This is because it acts on one level higher and thus is always applied first. For Min this means it splits it into two before the auxiliary variables are created, which means an unnecessary aux variable is made. I forget but there's another issue with it as well.

Basically I removed it since it was more trouble than it was worth, and fixing the issues wasn't the focus of this PR.

Reply:
This rule was new to this PR by the way, we've always just been using geq, leq, and sumgeq etc.

Reply:
Could we not keep this rule but at a lower priority level? I think you want `a=min(b,c)` to be handled by another rule, which is fine, but what about `a+b+c=d` etc?

Reply:
Eq has been reintroduced in the most recent commit! The Base rules now have a higher priority than Minion to ensure it though.